src/utils/api.js

```diff
@@ -4,9 +4,18 @@ const baseApi = axios.create({
   baseURL: "https://nc-project.onrender.com/api",
 });
 
-export async function getAllArticles() {
-  const result = await baseApi.get("/articles");
-  return result;
+export async function getAllArticles(topic) {
+  try {
+    if (topic) {
+      const result = await baseApi.get(`/articles?topic=${topic}`);
+      return result;
+    } else {
+      const result = await baseApi.get("/articles");
+      return result;
+    }
```

Comment:
Check out the notes on a nicer and easier way to handle params

https://notes.northcoders.com/courses/js-front-end/axios